scripts/autoPMM.py

```diff
@@ -0,0 +1,109 @@
+from typing import List
+from decimal import Decimal
+
+from hummingbot.core.data_type.order_candidate import OrderCandidate
+from hummingbot.core.utils.async_utils import safe_ensure_future
+from hummingbot.strategy.script_strategy_base import ScriptStrategyBase
+from hummingbot.core.data_type.common import OrderType, TradeType
+from hummingbot.core.rate_oracle.rate_oracle import RateOracle
+
+
+class AutoPMM(ScriptStrategyBase):
+    position_size = 1000
+    trading_pair = "AAVE-USDT"
+    exchange = "binance_paper_trade"
+    proposal: List[OrderCandidate] = []
+    proposal_adjusted: List[OrderCandidate] = []
+
+    markets = {exchange: {trading_pair}}
+
+    def on_tick(self):
+        try:
+            self.cancel_all_orders()
+            self.proposal= self.create_proposal()
+            self.proposal_adjusted = self.adjust_proposal_to_budget(self.proposal)
+            self.place_orders(self.proposal_adjusted)
+            
+        except Exception as e:
+            print(e)
+        # HummingbotApplication.main_application().stop()
+
+    def get_order_amount(self):
+        
+        conversion_rate = RateOracle.get_instance().rate(self.trading_pair)
+        amount = self.position_size / conversion_rate 
+        order_amount = self.connectors[self.exchange].quantize_order_amount(self.trading_pair, amount)
+        
+        self.logger().info(f"order_amount: {order_amount}") 
+        
+        return order_amount 
+
+            
+    def calculate_spread(self):
+        
+        mid_price = self.connectors[self.exchange].get_mid_price(self.trading_pair)
+        bid_price = self.connectors[self.exchange].get_price(self.trading_pair, False)
+        ask_price = self.connectors[self.exchange].get_price(self.trading_pair, True)
+
+        bid_ask_spread = (bid_price - ask_price )* mid_price / 100
```

Comment:
You take the absolute spread (bid - ask) and multiply it with 1% of mid-price? This will yield very different results, depending on mid-price alone. Higher mid-price = higher spread, which is incorrect.
Also, by subtracting the higher ask from the lower bid, will always result in a negative value for `bid_ask_spread`.

Reply:
Example with two markets, each having 1% bid and ask spread:
b(id) = 0.99, a(sk) = 1.01, (mid-)p(rice) = 1.00 => `-0.02 * 1/100 = -0.0002` => _0.02 % calculated spread_
b = 990, a = 1010, p = 1000 => `-20 * 1000/100 = -200` => _20 % calculated spread_

Reply:
```
 def calculate_spread(self):
        
        mid_price = self.connectors[self.exchange].get_mid_price(self.trading_pair)
        bid_price = self.connectors[self.exchange].get_price(self.trading_pair, is_buy = False)
        ask_price = self.connectors[self.exchange].get_price(self.trading_pair, is_buy = True)

        bid_ask_spread = (ask_price - bid_price)
        self.logger().info(f"best bid: {bid_price}")
        self.logger().info(f"best ask: {ask_price}")
        self.logger().info(f"bid_ask_spread in % : {bid_ask_spread * Decimal(100)} %")

        return bid_ask_spread

```

This should be fine.